pkg/apic/apiservice.go

```diff
@@ -130,7 +130,10 @@ func (c *ServiceClient) getAPIServiceByPrimaryKey(primaryKey string) (*v1alpha1.
 
 func (c *ServiceClient) getAPIServiceFromCache(serviceBody *ServiceBody) (*v1alpha1.APIService, error) {
 	if serviceBody.PrimaryKey != "" {
-		return c.getAPIServiceByPrimaryKey(serviceBody.PrimaryKey)
+		apiService, err := c.getAPIServiceByPrimaryKey(serviceBody.PrimaryKey)
+		if apiService != nil && err == nil {
+			return apiService, err
+		}
```

Comment:
do you not care if the get by primary key did return an error and at least log a warn?  like you tried but it failed so you used apiid?

Reply:
this is only for determining how to process the service, either create or update.  I do not think we need to log that something was not in the cache.

Reply:
ah, that makes sense...  ok, ignore.